main.cc

```diff
@@ -569,6 +569,12 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
 
             cfg->broadcast_to_all_shards().get();
 
+            // We pass this piece of config through a global as a temporary hack.
+            // See the comment at the definition of sstables::global_cache_index_pages.
+            smp::invoke_on_all([&cfg] {
+                sstables::global_cache_index_pages = cfg->cache_index_pages.operator utils::updateable_value<bool>();
```

Comment:
Is this operator call necessary? It's not marked explicit.

Reply:
It's not necessary per se, but I wanted to make explicit the fact that we are dealing with `updateable_value<bool>` rather than a `bool`.
E.g. if you typed `sstables::global_cache_index_pages = cfg->cache_index_pages;` it would work as intended, because `cfg->cache_index_pages` (a `named_value<bool>`) is implicitly convertible to `updateable_value<bool>`, but you could very easily confuse this with `sstables::global_cache_index_pages = cfg->cache_index_pages();`. `cfg->cache_index_pages()` is a `bool` and `updateable_value<T>` has `operator=(T)`, so this would still compile and set the flag correctly on startup, but would break the live updating silently.

Reply:
A standard way to do this would be via static_cast<>() rather than calling the operator.

The problem you point out seems to be at the API level, so perhaps needs an API-level solution if we acknowledge it. Those explicit operator calls are ugly. We could add `named_value::updateable_value()` member method for the conversion.